core/src/main/java/org/mule/runtime/core/internal/streaming/object/Bucket.java

```diff
@@ -14,36 +14,63 @@
 import java.util.Optional;
 
 /**
- * 
- * @param <T>
+ * An indexed group of items.
+ * <p>
+ * Buckets have a fixed capacity. Once reached, the bucket will accept no more items
+ *
+ * @param <T> the generic type of the items
+ * @since 4.0
  */
 public class Bucket<T> {
 
   private final List<T> items;
   private final int capacity;
   private final int index;
 
+  /**
+   * Creates a new instance
+   *
+   * @param index    the bucket's instance
+   * @param capacity the bucket's capacity.
+   */
   public Bucket(int index, int capacity) {
     this.index = index;
     this.capacity = capacity;
     this.items = new ArrayList<>(capacity);
   }
 
+  /**
+   * Obtains the value at the given index, if present
+   * @param index the item's index
+   * @return an {@link Optional} value
```

Comment:
if a value was added, it will be present. Otherwise, empty. I'll clarify it in the jdoc